floyd/cli/experiment.py

```diff
@@ -182,12 +182,30 @@ def get_log_id(job_id):
     return instance_log_id
 
 
+def check_job_termination_line(log_output_line, termination_list):
+    """
+    Get Job termination by inspecting if any of the termination log lines is detected
+    in the current log output line.
+    """
+    return any(terminal_line in log_output_line for terminal_line in termination_list)
+
+
 def follow_logs(instance_log_id, sleep_duration=1):
     cur_idx = 0
-    while True:
+    job_terminated = False
+
+    success = "[success] Finished execution"
+    failure = "[failed] Task execution failed"
+    shutdown = "[shutdown] Task execution cancelled"
+    timeout = "[timeout] Task execution cancelled"
+    termination_list = [success, failure, shutdown, timeout]
+
+    while not job_terminated:
         # Get the logs in a loop and log the new lines
         log_file_contents = ResourceClient().get_content(instance_log_id)
         print_output = log_file_contents[cur_idx:]
+        # Get the status of the Job from the current log line
+        job_terminated = check_job_termination_line(print_output, termination_list)
```

Comment:
This could be simplified as:

`job_terminated = print_output in termination_list`

Then you can remove the extra function call here.

Reply:
I cannot go for `job_terminated = print_output in termination_list`, because it's conditioning for a perfect match while in this case, I'm performing a substring match.